File: data/multilang_comparison/program_cpp_vector/src/func17.cpp
#include "program_cpp_vector.hpp" 
Array* func17(Array_param* vars, int loopsFactor) {
   size_t pCounter = vars->size;
   Array* array104;
   if (pCounter > 0) {
      array104 = vars->data.at(--pCounter);
      array104->refC++;
      DEBUG_COPY(array104->id);
   } else {
      array104 = new Array();
      array104->size = 269;
      array104->refC = 1;
      array104->id = 104;
      array104->data = std::vector<unsigned int>(array104->size, 0);
      DEBUG_NEW(array104->id);
   }
   Array_param params0;
   params0.size = 1;
   std::vector<Array*> data_params0(1);
   params0.data = data_params0;
   params0.data.at(0) = array104;
   Array* array105 = func24(&params0, get_path(), loopsFactor);
   DEBUG_RETURN(array105->id);
   if(params0.size > 0) {
   	 params0.data.clear();
   }
   unsigned int loop31 = 0;
   unsigned int loopLimit31 = (50)/3 + 1;
   for(; loop31 < loopLimit31; loop31++) {
      for (int i = 0; i < array104->size; i++) {
         array104->data.at(i)++;
      }
   }
   for (int i = 0; i < array104->size; i++) {
       if (array104->data.at(i) == 92) {
           return array104;
       }
   }
   array105->refC--;
   if(array105->refC == 0) {
      array105->data.clear();
      delete array105;
      DEBUG_FREE(array105->id);
   }
   return array104;
}

File: data/multilang_comparison/program_go/src/func17.go
package main
import "math/rand"
var _ = rand.Intn
func Func17(vars *ArrayParam, loopsFactor int) *Array{
   var pCounter int = vars.Size;
   _ = pCounter
   var array104 *Array
   if pCounter > 0 {
       array104 = vars.Data[pCounter-1]
       pCounter--
       array104.RefC++
   } else {
       array104 = &Array{
           Size: 269,
           RefC: 1,
           Id: 104,
           Data: make([]uint32, 269),
       }
   }
   var params0 ArrayParam
   params0.Size = 1
   params0.Data = make([]*Array, params0.Size)
   params0.Data[0] = array104
   var array105 *Array = Func24(&params0, getPath(), loopsFactor);
   params0.Data = nil
   params0.Size = 0
   var loop31 int = 0;
   var loopLimit31 int = (50)/3 + 1;
   for ; loop31 < loopLimit31; loop31++ {
      for i := 0; i < array104.Size; i++ {
          array104.Data[i]++
      }
   }
   for i := 0; i < array104.Size; i++ {
       if array104.Data[i] == 92 {
           return array104
       }
   }
   array105.RefC--
   if array105.RefC == 0 {
       array105.Data = nil
       array105 = nil
   }
   return array104;
}

